DEVELOPING.md

```diff
@@ -179,72 +192,282 @@ kubectl get pod -n ambassador
 
-1. **Identify an  Issue**
-   - Start by creating or searching for an open issue (*it is good to check open PR's too*)
-   - We will assign an issue to you so that others know who is working on it
+### Submitting a Pull Request (PR)
 
-2. **Cut a development branch from `master`**.
-   - We recommend that your branches start with your username.
-      - At Ambassador Labs we typically use `git-flow`-style naming, e.g. `flynn/dev/telepathic-ui`
-   - Please do not use a branch name starting with `release`.
+> If you haven't set up your development environment then please see the [Development Setup](#development-setup) section.
 
-3. If your development takes any significant time, **merge master back into your branch regularly**.
-   - Think "every morning" and "right before submitting a pull request."
-   - If you're using a branch name that starts with your username, `git rebase` is also OK and no one from Ambassador Labs will scream at you for force-pushing.
-   - Please do **not** rebase any branch that does **not** start with your username.
+When submitting a Pull Request (PR) here are a set of guidelines to follow:
 
-   ```bash
-   # make sure you local master branch has the latest
-   git checkout master
-   git pull -ff origin master
+1. Search for an [existing issue](https://github.com/emissary-ingress/emissary/issues) or create a [new issue](https://github.com/emissary-ingress/emissary/issues/new/choose).
+
+2. Be sure to describe your proposed change and any open questions you might have in the issue. This allows us to collect historical context around an issue, provide feedback on the proposed solution and discuss what versions a fix should target.
+
+3. If you haven't done so already create a fork of the respository and clone it locally
 
-   # rebase development branch on master
-   # fix any conflicts if they occur during rebase
-   git checkout <your-dev-branch>
-   git rebase master
+   ```shell
+   git clone <your-fork>
+   ```
 
-   # if you previously pushed your changes to Github then your rebase will require you to force push
-   git push origin <your-dev-branch> -f
+4. Cut a new patch branch from `master`:
+
+   ```shell
+   git checkout master
+   git checkout -b my-patch-branch master
    ```
 
-4. **Code changes must have associated documentation updates.**
-   - Make changes in <https://github.com/datawire/ambassador-docs> as necessary,
-   and include a reference to those changes the pull request for your code
-   changes.
-   - See [Contributing to Docs](#contributing-to-the-docs) for more details.
+5. Make necessary code changes.
 
-5. **Code changes must include passing tests.**
-   - See `python/tests/README.md` for more here.
-   - Your tests **must** actually test the change you're making.
-   - Your tests **must** pass in order for your change to be accepted.
+   - Make sure you include test coverage for the change.
+   - Ensure code linting is passing
```

Comment:
maybe go into detail on how to run tests and lint. getting the full environment setup to run tests locally is somewhat involved.

Esp because as of right now community members can't run CI

Reply:
Since this is more less a quick check list I will point at the current testing section and add a note about running `make lint`.

If we want to improve the instructions in those sections, we can PR those separate.